atlasdb-cassandra/src/main/java/com/palantir/atlasdb/cassandra/CassandraAtlasDbFactory.java

```diff
@@ -106,27 +100,32 @@ private static CassandraKeyValueServiceConfig toCassandraConfig(KeyValueServiceC
     }
 
     @VisibleForTesting
-    Supplier<CassandraKeyValueServiceRuntimeConfig> preprocessKvsRuntimeConfig(
-            Supplier<Optional<KeyValueServiceRuntimeConfig>> runtimeConfig) {
-        return () -> {
-            Optional<KeyValueServiceRuntimeConfig> configOptional = runtimeConfig.get();
-
-            return configOptional
-                    .map(config -> {
-                        if (!(config instanceof CassandraKeyValueServiceRuntimeConfig)) {
-                            log.error(
-                                    "Invalid KeyValueServiceRuntimeConfig. Expected a KeyValueServiceRuntimeConfig of"
+    Refreshable<CassandraKeyValueServiceRuntimeConfig> preprocessKvsRuntimeConfig(
+            Refreshable<Optional<KeyValueServiceRuntimeConfig>> runtimeConfig) {
+        return runtimeConfig.map(maybeConfig -> maybeConfig
+                .map(config -> {
+                    if (!(config instanceof CassandraKeyValueServiceRuntimeConfig)) {
+                        log.error(
+                                "Invalid KeyValueServiceRuntimeConfig. Expected a KeyValueServiceRuntimeConfig of"
                                         + " type CassandraKeyValueServiceRuntimeConfig, found {}. Using latest valid"
                                         + " CassandraKeyValueServiceRuntimeConfig.",
-                                    SafeArg.of("configClass", config.getClass()));
-                            return latestValidRuntimeConfig;
-                        }
-
-                        latestValidRuntimeConfig = (CassandraKeyValueServiceRuntimeConfig) config;
+                                SafeArg.of("configClass", config.getClass()));
                         return latestValidRuntimeConfig;
-                    })
-                    .orElseGet(CassandraKeyValueServiceRuntimeConfig::getDefault);
-        };
+                    }
+                    latestValidRuntimeConfig = (CassandraKeyValueServiceRuntimeConfig) config;
+                    return latestValidRuntimeConfig;
+                })
+                .orElseGet(CassandraKeyValueServiceRuntimeConfig::getDefault));
+    }
+
+    @VisibleForTesting
+    CassandraKeyValueServiceConfig getConfigWithNamespace(KeyValueServiceConfig config, Optional<String> namespace) {
```

Comment:
This came with some prescient knowledge - once we switch over to the merged config providing a runtime config, then tests can't use `createMergedKeyValueServiceConfig` to test whether the install config has a namespace applied. 

Separating this out also just seems a little cleaner, imo.